dev/tests/integration/testsuite/Magento/GoogleAnalytics/Block/GaTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\GoogleAnalytics\Block;
+
+use Magento\Framework\App\ObjectManager;
+use Magento\TestFramework\TestCase\AbstractController;
+
+class GaTest extends AbstractController
+{
+    /**
+     * Layout instance
+     *
+     * @var \Magento\Framework\View\LayoutInterface
+     */
+    private $layout;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->dispatch('/');
+        $this->layout = ObjectManager::getInstance()->get(
+            \Magento\Framework\View\LayoutInterface::class
+        );
+    }
+
+    /**
+     * Test if block exists in head tag.
+     *
+     * @magentoAppArea frontend
+     * @magentoConfigFixture current_store google/analytics/active 1
+     * @magentoConfigFixture current_store google/analytics/account XXXXXXX
+     */
+    public function testBlockIsInHead()
+    {
+        $this->assertNotNull(
+            $this->layout->getChildBlock('header.container', 'google_analytics')
+        );
+    }
+
+    /**
+     * Test if block does not exist in body tag.
+     *
+     * @magentoAppArea frontend
+     * @magentoConfigFixture current_store google/analytics/active 1
+     * @magentoConfigFixture current_store google/analytics/account XXXXXXX
+     */
+    public function testBlockIsNotInBody()
```

Comment:
Please combine with previous test.

Reply:
@orlangur Hello, thanks for your review. I think, that making  one assert per test method is best practice, because making few assertion together here automatically makes test condition (its semantic, name) more generic. From my point, multiple aspects of the outcome of some action have to be tested separately, and the tests should be able to succeed or fail independent of each other.

Reply:
> making one assert per test method is best practice

What is the ground for such statement?

There is no sense to have duplicate tests which differ only in asserts from both maintenance and performance perspective.

Reply:
@orlangur the reason is defects localization. It is a well known recommendation to have a single assertion per unit test. You may read more on the subject at http://xunitpatterns.com/Principles%20of%20Test%20Automation.html#Single%20Condition%20Test

Reply:
@vkublytskyi, 

> There is a strong temptation to reuse the end state of one test condition as the starting state of the next by combining them into a single test because this makes it more efficient. This is not recommended because when one assertion fails, **the rest of the test is not executed** . This makes it hard to achieve Defect Localization.

> Some test drivers insist on one assertion per test
> Having one assertion per test makes such naming very easy but it does lead to many more test methods if we have to assert on many output fields. Of course, we can often comply with this interpretation by extracting a Custom Assertion (page X) or Verification Method (see Custom Assertion) that allows us to reduce the multiple assertion method calls into one. **Sometimes that makes the test more readable but when it doesn't, I wouldn't be too dogmatic about insisting on a single assertion.**

In our case there is no test execution at all, only preconditions. So SUT has One Condition by design and it is totally fine to perform as many assertions as we need.